libs/blocks/aside/aside.js

```diff
@@ -111,7 +111,43 @@ function checkViewportPromobar(foreground) {
   if (childCount < 3) addPromobar(children[childCount - 1], foreground);
 }
 
-function combineTextBocks(textBlocks, iconArea, viewPort, variant) {
+function toolTipPosition(container) {
+  const isRtl = document.documentElement.getAttribute('dir') === 'rtl';
+  const isTablet = container.classList.contains('tablet-up');
+  const isMobile = container.classList.contains('mobile-up');
+  if ((isRtl && isTablet) || (isMobile && !isRtl)) {
+    return 'right';
+  }
+  return 'left';
+}
+
+async function addTooltip(foreground) {
+  const desktopContentText = foreground.querySelector('.desktop-up .text-area')?.textContent.trim();
+  const toolTipIcons = [];
+  [...foreground.querySelectorAll(':scope > div:not(.desktop-up)')].forEach((viewPortEl) => {
+    const childContent = viewPortEl.querySelector('.text-area');
+    const childContentText = childContent.textContent.trim();
+    if (childContentText === desktopContentText) return;
+    const appendTarget = viewPortEl.querySelector('.text-area').lastElementChild;
+    const iconWrapper = createTag('em');
+    iconWrapper.style.display = 'none';
+    iconWrapper.textContent = `${toolTipPosition(viewPortEl)}|${desktopContentText}`;
+    const tooltipSpan = createTag('span', { class: 'icon icon-tooltip' });
```

Comment:
Should the tooltip be visible on focus? I see this is the behaviour for tooltips in table - https://main--milo--adobecom.aem.page/docs/library/kitchen-sink/table